src/hera/workflows/_mixins.py

```diff
@@ -579,13 +579,18 @@ def _build_arguments(self) -> Optional[ModelArguments]:
         elif isinstance(self.arguments, ModelArguments):
             return self.arguments
 
+        from hera.workflows.workflow_template import WorkflowTemplate
+
         def add_argument(k: str, v: Any, result: ModelArguments):
+            is_workflow_template = isinstance(self, WorkflowTemplate)
+
             if isinstance(v, Parameter):
-                value = v.with_name(k).as_argument()
+                param_copy = v.with_name(k)
+                value = param_copy.as_input() if is_workflow_template else param_copy.as_argument()
```

Comment:
Can we do this via an override rather than an isinstance check? Maybe a `param_as_argument` function?

Reply:
As we already have the `as_argument` function on `Parameter`, I don't think there's a much cleaner solution here. As it's a single special case I will leave it as is and refactor if needed in future